csharp/ql/lib/semmle/code/csharp/TypeRef.qll

```diff
@@ -12,7 +12,13 @@ private class TypeRef extends @typeref {
 
   string toString() { result = this.getName() }
 
-  Type getReferencedType() { typeref_type(this, result) }
+  Type getReferencedType() {
+    typeref_type(this, result)
+    or
+    not typeref_type(this, _) and
+    result instanceof UnknownType and
+    types(result, _, "<unknown type>")
```

Comment:
Isn't `not typeref_type(this, _) and result instanceof UnknownType` enough?

Reply:
Yes, we should only ever have one `UnknownType` entity.